homeassistant/components/tessie/climate.py

```diff
@@ -0,0 +1,129 @@
+"""Climate platform for Tessie integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from tessie_api import (
+    set_climate_keeper_mode,
+    set_temperature,
+    start_climate_preconditioning,
+    stop_climate,
+)
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import TessieDataUpdateCoordinator
+from .entity import TessieEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Tessie Climate platform from a config entry."""
+    coordinators = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(TessieClimateEntity(coordinator) for coordinator in coordinators)
+
+
+class TessieClimateEntity(TessieEntity, ClimateEntity):
+    """Vehicle Location Climate Class."""
+
+    _attr_precision = 0.5
+    _attr_min_temp = 15
+    _attr_max_temp = 28
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_hvac_modes = [HVACMode.HEAT_COOL, HVACMode.OFF]
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.PRESET_MODE
+    )
+    _attr_preset_modes: list = ["off", "on", "dog", "camp"]
+
+    def __init__(
+        self,
+        coordinator: TessieDataUpdateCoordinator,
+    ) -> None:
+        """Initialize the Climate entity."""
+        super().__init__(coordinator, "climate")
+
+    @property
+    def hvac_mode(self) -> HVACMode | None:
+        """Return hvac operation ie. heat, cool mode."""
+        if self.get("climate_state_is_climate_on"):
+            return HVACMode.HEAT_COOL
+        return HVACMode.OFF
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return the current temperature."""
+        return self.get("climate_state_inside_temp")
+
+    @property
+    def target_temperature(self) -> float | None:
+        """Return the temperature we try to reach."""
+        return self.get("climate_state_driver_temp_setting")
+
+    @property
+    def max_temp(self) -> float:
+        """Return the maximum temperature."""
+        return self.get("climate_state_max_avail_temp", self._attr_min_temp)
+
+    @property
+    def min_temp(self) -> float:
+        """Return the minimum temperature."""
+        return self.get("climate_state_min_avail_temp", self._attr_max_temp)
```

Comment:
Did you switch around the fallback values?

Reply:
Yes.... yes I did.

Reply:
Hehe luckily we have reviews